cimsparql/model.py

```diff
@@ -397,6 +397,35 @@ def borders(
             return self.client.query_with_header(query, limit)
         return self.get_table_and_convert(query, limit, index="mrid")
 
+    def exchange(
+        self,
+        region: Union[str, List[str]],
+        nodes: str,
+        ignore_hvdc: bool = True,
+        with_market_code: bool = False,
+        market_optional: bool = False,
```

Comment:
**Similar**: Both with or without market and if it should be optional --> Three cases to test

1. `with_market_code=True`, `market_optional=False`
2. `with_market_code=True`, `market_optional=True`
3. `with_market_code=False

I think if `market` can be written without making explicit acess to the SN namespace both argument can be completely removed. But for the time being I guess we have to keep `with_market_code`